### ISSUE TITLE
Georeferencer takes far longer than copied command in CLI

### ISSUE THREAD

**Issue by roydenyates** [qgis/QGIS#52349]
### What is the bug or the crash?

Georeferencer in QGIS: 270 secs
GDAL CLI: 25 secs

Extreme (fringe) use case, but it might have exposed an issue.

### Steps to reproduce the issue

Using georeferencer, 2000+ GCPs, thin plate spline.  Setup saved file name etc, in settings, then "start".  Time completion.

Same raster and GCPs, use "generate GDAL script", copy and go to folder with raster, in terminal, run the copied commands and time completion.

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.31.0-Master | QGIS code revision | b8458777cd
-- | -- | -- | --
Qt version | 5.15.6
Python version | 3.10.7
GDAL/OGR version | 3.5.1
PROJ version | 9.0.1
EPSG Registry database version | v10.064 (2022-05-19)
GEOS version | 3.11.0-CAPI-1.17.0
SQLite version | 3.39.3
PostgreSQL client version | unknown
SpatiaLite version | 5.0.1
QWT version | 6.1.4
QScintilla2 version | 2.11.6
OS version | Ubuntu 22.10
  |   |   |  
This copy of QGIS writes debugging output.
  |   |   |  
Active Python plugins
maplibrary | 1.2
searchlayers | 3.0.12
qfieldsync | v4.4.9
processing_wbt | 1.3.6
qgis_resource_sharing | 1.0.0
processing_r | 3.1.1
compass_routes | 0.3.1
DigitizingTools | 1.5.1
latlontools | 3.6.9
moving_feature | 2.0
enmapboxplugin | 3.12-alpha.9
qlyrx | 0.3.3
slyr_community | 4.0.7
otbprovider | 2.12.99
db_manager | 0.1.20
processing | 2.12.99
grassprovider | 2.12.99

</body></html>QGIS version
3.31.0-Master
QGIS code revision
[b8458777cd](https://github.com/qgis/QGIS/commit/b8458777cd)
Qt version
5.15.6
Python version
3.10.7
GDAL/OGR version
3.5.1
PROJ version
9.0.1
EPSG Registry database version
v10.064 (2022-05-19)
GEOS version
3.11.0-CAPI-1.17.0
SQLite version
3.39.3
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.4
QScintilla2 version
2.11.6
OS version
Ubuntu 22.10

This copy of QGIS writes debugging output.

Active Python plugins
maplibrary
1.2
searchlayers
3.0.12
qfieldsync
v4.4.9
processing_wbt
1.3.6
qgis_resource_sharing
1.0.0
processing_r
3.1.1
compass_routes
0.3.1
DigitizingTools
1.5.1
latlontools
3.6.9
moving_feature
2.0
enmapboxplugin
3.12-alpha.9
qlyrx
0.3.3
slyr_community
4.0.7
otbprovider
2.12.99
db_manager
0.1.20
processing
2.12.99
grassprovider
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

Tried Settings -> Options -> Processing -> Max Threads and increasing that, but no difference.

**Comment by roydenyates** [comment]
Further info:

I now realise that my tests with gdalwarp at the CLI utilise just 1cpu per warp.

With options "-multi  -wo NUM_THREADS=ALL_CPUS" it is even faster  by much.

**Comment by oznogon** [comment]
**tl;dr:** Even with the added overhead of copying/pasting to the command line and manually updating the raster layer in QGIS after georeferencing, running the generated gdalwarp command on the command line — with _or without_ adding multi-core processing flags — saves me more than 10 minutes per iteration over running the georeferencing step in the GUI, with no difference in quality.

Input:

- 32.5MB uncompressed 2517x3272 TIFF source image
- 938 GCPs

Hardware:

- Ryzen 5 7840U (6 cores/12 threads) CPU
- 32GB dual-channel non-ECC RAM
- 2TB WD Black SN850X NVMe SSD

Software:

- Fedora 40, KDE Plasma DE on Wayland
- QGIS 3.36.1 installed via Flathub (KDE Flatpak runtime; 3.36.2 not yet available)
- Python 3.11.9
- GDAL 3.8.5 (2024/04/02)

QGIS Georeferencer GUI transformation settings:

- Tranformation type: Thin Plate Spline
- Target CRS: EPSG:4326 - WGS 84
- Resampling method: Lanczos (6x6 Kernel)
- Compression: None
- Create world file only (linear transforms): false
- Use 0 for transparency when needed: false
- Set target resolution: false
- Generate PDF map: none
- Generate PDF report: none
- Save GCP points: false
- Load in project when done: true

QGIS Georeferencer-generated GDAL script, with GCPs truncated for brevity:

```
gdal_translate -of GTiff -gcp 2137.007 3227.588 178.7889471 -32.98137181 ... -gcp 2256.991 2491.122 182.67034811 -10.86439727 "/home/.../map.tif" "/tmp/map.tif"
gdalwarp -r lanczos -tps -co COMPRESS=None  -t_srs EPSG:4326 "/tmp/map.tif" "/home/.../map_modified.tif"
```

gdalwarp command with multi-core flags, per @roydenyates:

```
gdalwarp -r lanczos -tps -co COMPRESS=None  -t_srs EPSG:4326 -multi -wo NUM_THREADS=ALL_CPUS "/tmp/map.tif" "/home/.../map_modified.tif"
```

Results:

| Method | Real time | CPU cores used | % time taken vs. georeferencing via GUI |
|-|-|-|-|
| QGIS GUI "Start Georefrencing" output | 12m1s | 1 core | — |
| "Generate GDAL script" output | 1m21s | 1 core | 11.25% (8.9x) |
| "Generate GDAL script" output with `-multi -wo NUM_THREADS=ALL_CPUS` | 0m10s | 12 cores | 1.4% (72x)

These results are consistently reproducible in this project. The command-line output is visually identical in QGIS to the output from the Georeferencer GUI and has identical file sizes. The GUI-generated result has a different MD5 checksum than the single-/multi-core gdalwarp outputs, which are identical to each other.

**Comment by oznogon** [comment]
Informally, no difference in times using 3.34.3 Georeferencer GUI either.

**Comment by agiudiceandrea** [comment]
@oznogon, so does the issue no longer occur on your system using QGIS 3.34?
